File: app/src/main/java/cc/lixiaoyu/wanandroid/util/DBHelper.java
package cc.lixiaoyu.wanandroid.util;

import java.util.List;

/**
 * 数据库操作接口，提供应用中需要数据库存储的的方法
 */
public interface DBHelper {
    /**
     * 添加一个搜索历史数据
     * @param historyData
     * @return
     */
    boolean addHistoryData(String historyData);

    /**
     * 删除一条搜索历史
     * @param historyData
     * @return
     */
    void deleteHistoryData(String historyData);

    /**
     * 清空所有的搜索历史
     * @return
     */
    void clearAllHistoryData();

    /**
     * 加载所有的搜索历史
     * @return
     */
    List<String> loadAllHistoryData();
}

File: app/src/main/java/cc/lixiaoyu/wanandroid/util/storage/DBHelper.kt
package cc.lixiaoyu.wanandroid.util.storage

import cc.lixiaoyu.wanandroid.entity.HistoryData
import org.litepal.LitePal

/**
 * DBHelper实现类
 */
class DBHelper {
    fun addHistoryData(historyData: String?): Boolean {
//        //先查重
//        List<HistoryData> result = LitePal.where("data=?", historyData)
//                .find(HistoryData.class);
//        if(result == null){
//            //使用Litepal保存数据
        val hd = HistoryData(historyData!!)
        return hd.save()
        //        }
//        return false;
    }

    fun deleteHistoryData(historyData: String?) {
        LitePal.deleteAll(HistoryData::class.java, "data=?", historyData)
    }

    fun clearAllHistoryData() {
        LitePal.deleteAll(HistoryData::class.java)
    }

    fun loadAllHistoryData(): List<String> {
        val hdList = LitePal.findAll(
            HistoryData::class.java
        )
        val list: MutableList<String> = ArrayList(hdList.size)
        for (hd in hdList) {
            list.add(hd.data)
        }
        return list
    }
}